pkg/roachprod/install/cockroach.go

```diff
@@ -882,8 +903,18 @@ func (c *SyncedCluster) generateStartArgs(
 
 	// if neither --log nor --log-config-file are present
 	if idx1 == -1 && idx2 == -1 {
-		// Specify exit-on-error=false to work around #62763.
-		args = append(args, "--log", `file-defaults: {dir: '`+logDir+`', exit-on-error: false}`)
+		loggingConfigFile, err := execLoggingTemplate(loggingTemplateData{
+			LogDir: logDir,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed rendering logging template")
+		}
+
+		if err := c.PutString(ctx, l, c.Nodes, loggingConfigFile, "cockroachdb-logging.yaml", 0644); err != nil {
```

Comment:
This may have unintended consequences for virtual clusters, as the `logDir` above is unique for each virtual cluster, but the same config is going to be overwritten for each virtual cluster started.

Reply:
Are you saying that in the case of virtual clusters that there should be a single logging configuration? I'm not familiar with the configuration layout of virtual clusters so perhaps I'm missing a bit of context.

Reply:
The `loggingTemplateData` will contain the logDir, which is different for each virtual cluster. But it overrides the same file and passes the same config. In other words I think maybe the config file should also encode the virtual cluster info to avoid overriding the same config.

As it is it probably won't cause issues because the file gets loaded into memory I presume, so if you start a virtual cluster on the same node after if the file gets overridden it won't matter as much.

```
// LogDir returns the logs directory for the given node.
func (c *SyncedCluster) LogDir(node Node, virtualClusterName string, instance int) string {
	dirName := "logs" + virtualClusterDirSuffix(virtualClusterName, instance)
	if c.IsLocal() {
		return filepath.Join(c.localVMDir(node), dirName)
	}
	return dirName
}
```

To me it might make sense to encode virtualClusterName and instance into the loggingConfigFile name as well.

Reply:
Updated to use the `virtualClusterDirSuffix` in the file name.